### PR TITLE
add rat icon

### PR DESCRIPTION
**Pull request by henri42**
Here is a little rat icon.
I hope it respects the guidelines correctly !

rat

### CODE REVIEW

File: icons/rat.svg [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    stroke="currentColor"
+    stroke-width="2"
+    stroke-linecap="round"
+    stroke-linejoin="round"
+>    
+    <path d="M7.185,12.815L3,17C3,17 2.781,14.817 3,12.5C3.201,10.372 3.804,8.325 3.777,8.301C3.339,8.116 2.94,7.821 2.625,7.421C1.629,6.156 1.847,4.32 3.112,3.324C4.377,2.327 6.213,2.546 7.209,3.811"/>
+    <path d="M7,4.846C7.011,4.371 7.139,3.891 7.395,3.451C8.204,2.059 9.991,1.586 11.383,2.396C12.775,3.205 13.248,4.992 12.439,6.384C12.388,6.472 12.333,6.556 12.274,6.636C17.78,7.49 22,12.256 22,18C22,19.407 21.747,20.755 21.284,22.001C21.216,22.001 5.653,22 5.653,22"/>
+    <path d="M8.635,22.001C8.327,21.318 8.155,20.56 8.155,19.763C8.155,16.752 10.6,14.308 13.61,14.308C14.253,14.308 14.871,14.419 15.444,14.624"/>
+    <path d="M22,18.154C22,13.408 20.566,9.001 18.073,5.355C17.839,5.011 17.702,4.596 17.702,4.149C17.702,2.963 18.665,2 19.851,2C21.037,2 22,2.963 22,4.149C22,4.496 21.918,4.823 21.771,5.114"/>
+    <path d="M7.783 8.828v.0"></path>
```
Comment (jguddas) [inline icons/rat.svg:17]:
```suggestion
        <path d="M7.73 8.78H7.74"/>
```

File: icons/rat.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "$schema": "../icon.schema.json",
+  "tags": [
+    "animal",
+    "mouse"
```
Comment (jguddas) [inline icons/rat.json:5]:
```suggestion
    "mouse",
    "mice",
    "rodent",
    "pet",
    "pest",
    "plague"
```

**Comment by jguddas** [comment]
Should be 4px IMO.

File: icons/rat.svg (hunk 1 above)
Comment (karsa-mistmere) [inline icons/rat.svg:17]:
```suggestion
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>    
  <path d="M7.19 12.81 3 17s-.22-2.18 0-4.5c.2-2.13.8-4.18.78-4.2A2.92 2.92 0 1 1 7.2 3.81" />
  <path d="M7 4.85a2.92 2.92 0 1 1 5.27 1.79A11.5 11.5 0 0 1 21.28 22H5.65" />
  <path d="M8.64 22a5.46 5.46 0 0 1 6.8-7.38" />
  <path d="M22 18.15c0-4.74-2.55-8.63-3.82-12.82A4.1 4.1 0 0 1 18 4a2 2 0 1 1 4 0" />
  <path d="M8 9h.01" />
```

Changes made:
- fixed indenting (2 spaces of indent)
- fixed the SVG root element to match other files
- added a 2px gap to the tail
- made the eye .01px tall so it does't get removed in Figma and some other editors
- ran it through svgo to optimize the paths

**Comment by karsa-mistmere** [comment]
I also updated my original `mouse` design using parts of this icon to match, not sure it's worth adding anymore though:
image

**Comment by jguddas** [comment]
I love the mouse icons as well, I do feel like the tail is connecting weirdly tho.

File: icons/rat.svg (hunk 1 above)
Comment (jguddas) [reply icons/rat.svg:17]: So the tail is properly connected to the body
```diff
-  <path d="M22 18.15c0-4.74-2.55-8.63-3.82-12.82A4.1 4.1 0 0 1 18 4a2 2 0 1 1 4 0" />
+  <path d="M22 18c0-4.74-2.55-8.48-3.82-12.67A4.1 4.1 0 0 1 18 4c0-2.667 4-2.667 4 0"/>
```

**Comment by karsa-mistmere** [comment]
That's my older design that has guideline violations, but I think it looks pretty silly if connected the way you've illustrated:
image

File: icons/rat.svg (hunk 1 above)
Comment (jguddas) [reply icons/rat.svg:17]: Weird corner around at start of the nose fix
```diff
- <path d="M7.19 12.81 3 17s-.22-2.18 0-4.5c.2-2.13.8-4.18.78-4.2A2.92 2.92 0 1 1 7.2 3.81" />
+ <path d="M7.19 12.81 3 17s-.22-2.18 0-4.5c.2-2.13.78-4.2.78-4.2A2.92 2.92 0 1 1 7.2 3.81" />
```

Comment (karsa-mistmere) [reply icons/rat.svg:17]: @jguddas: I was struggling to see what exactly changed, I had to make a diff: :D
image

The nose is fine, but you've introduced some squaring at the tail's loop.

**Comment by jguddas** [comment]
You are right, it doesn't help, I just feel like it doesn't look like a tail and more like a ledge the mouse is sitting on.

It's super cute tho even with it looking like a ledge :)

**Comment by jguddas** [comment]
What about making the nose rounder?

**Comment by karsa-mistmere** [comment]
> I just feel like it doesn't look like a tail and more like a ledge the mouse is sitting on.

I think an easy fix for that would be shortening the bottom part:
image

**Comment by karsa-mistmere** [comment]
I'm fine with adding a little bit of rounding, it doesn't make a drastic difference on 100% either way.

File: icons/rat.svg (hunk 1 above)
Comment (jguddas) [reply icons/rat.svg:17]: I just moved the bottom node of the tail a couple of pixels up.
The nose had a Bézier handler that was basically at the node center, it introduced some weird clipping in inkscape for me and was unnecessary.

**Comment by henri42** [comment]
Thank you for the reviews, i think this last icon proposed by @karsa-mistmere is a good mix :
image

The tail is way better than the one from my design (mine is more looking like a cat tail). Moreover, I think it looks like both rat and mouse so it can be enough. 
So maybe we can only keep this one ?

**Comment by ericfennis** [comment]
@henri42 So to go further, we will add the mouse/rat icon from @karsa-mistmere? or..

**Review by karsa-mistmere — APPROVED** [review]

**Pull request merged**